### PR TITLE
Add test for single node network

### PR DESCRIPTION
**Pull request by LuKks**
Depends on https://github.com/mafintosh/dht-rpc/pull/57

### CODE REVIEW

**Comment by LuKks** [comment]
Will leave here some invalid tests in case it's useful in the future. They're invalid because two nodes are required for pinging each other to get their own stable address back, etc. Although, some changes can be done to make it work but it's probably not intended anyway

<details>
<summary>Tests</summary>

```js
test.skip('first persistent node with no host given', async function (t) {
  const bootstrap1 = new DHT({ bootstrap: [], ephemeral: false, firewalled: false })
  await bootstrap1.ready()
  t.is(bootstrap1.address().host, '::')

  t.ok(await makeServerAndClient([{ host: localIP(), port: bootstrap1.address().port }]))

  t.absent(await makeServerAndClient([{ host: '127.0.0.1', port: bootstrap1.address().port }]))

  await bootstrap1.destroy()
})

test.skip('first persistent node but binds to IPv6 localhost', async function (t) {
  // Note: anyway will not work due peer.id ipv4 encoding
  const bootstrap1 = new DHT({ bootstrap: [], ephemeral: false, firewalled: false, host: '::1' })
  await bootstrap1.ready()
  t.is(bootstrap1.address().host, '::1')

  t.ok(await makeServerAndClient([{ host: '::1', port: bootstrap1.address().port }]))

  await bootstrap1.destroy()
})

test.skip('first persistent node but binds to IPv4 localhost', async function (t) {
  const bootstrap1 = new DHT({ bootstrap: [], ephemeral: false, firewalled: false, host: '127.0.0.1' })
  await bootstrap1.ready()
  t.is(bootstrap1.address().host, '127.0.0.1')

  t.ok(await makeServerAndClient([{ host: '127.0.0.1', port: bootstrap1.address().port }]))

  await bootstrap1.destroy()
})

test.skip('first persistent node with no host given but binds to local host address', async function (t) {
  const bootstrap1 = new DHT({ bootstrap: [], ephemeral: false, firewalled: false, host: localIP() })
  await bootstrap1.ready()
  t.is(bootstrap1.address().host, localIP())

  t.ok(await makeServerAndClient([{ host: localIP(), port: bootstrap1.address().port }]))

  await bootstrap1.destroy()
})

test.skip('ephemeral node enters persistent mode, and later goes back to ephemeral', async function (t) {
  // This should test when a node is adaptive and becomes persistent naturally
  // but then later goes back into 'ephemeral' mode (i.e. due sleep, etc)
  // Currently, hyperdht is not destroying the persistent class object when it goes back to eph
})
```
</details>

File: test/bootstrapper.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const test = require('brittle')
+const { toArray } = require('./helpers')
+const DHT = require('../')
+const UDX = require('udx-native')
+
+test('single node network is enough to find peers', async function (t) {
+  const bootstrap1 = DHT.bootstrapper(49737, '127.0.0.1')
```
Comment (mafintosh) [inline test/bootstrapper.js:7]: is this expected that port to be avail?

**Comment by LuKks** [comment]
Depends on https://github.com/mafintosh/dht-rpc/pull/58

**Comment by mafintosh** [comment]
Just cleaning up, reopen if relevant still

**Pull request closed without merging**